readthedocs/projects/models.py

```diff
@@ -920,6 +921,13 @@ def conf_dir(self, version=LATEST):
 
     @property
     def has_good_build(self):
+        # Check if there is `_latest_successful_build` attribute in the Queryset.
+        # Used for database optimization.
+        if hasattr(self, self.LATEST_SUCCESSFUL_BUILD_CACHE):
+            if build_successful := getattr(self, self.LATEST_SUCCESSFUL_BUILD_CACHE):
```

Comment:
Same feedback about not needing the first `if`.

Reply:
Noted on #11616 but `hasattr` does indeed seem needed as `hasattr() == False` is what signals that prefetching is not in use and the normal method query can be run.